### ISSUE TITLE
Crash with MacOS Ventura + Z Fold 3

### ISSUE THREAD

**Issue by hungchieu** [ganeshrvel/openmtp#271]
Not sure it works on MacOS Ventura with other phones or not? But I'm facing OpenMTP Helper crashing with Samsung Z Fold 3 + MacOS Ventura.

**Comment by hungchieu** [comment]
[OpenMTP-log.zip](https://github.com/ganeshrvel/openmtp/files/9882647/OpenMTP-log.zip)
Here is the log file.

**Comment by somsak** [comment]
Same here! My phone is Samsung Galaxy Z Fold 4, though.

OpenMTP version: 3.1.15 (3.1.15)
MacOS version: 13.0 (Ventura) on Macbook Air M1 (2020)

Android version: 12

I already tried changing "Allow accessories to connect" under "Privacy & Security" to either "Automatically When Unlocked" and Always, but still not working.


BTW how can I get the log file? The "Report a bug" menu also show error about report generation failed.

**Comment by Githerazi** [comment]
Think it is a Ventura issue, openmtp not working for me on ventura with a Pixel3a xl running LineageOS. Android File Transfer not working either...

**Comment by ganeshrvel** [comment]
Could try these?

 1) Reboot your Mac (IMPORTANT)

 2) Do you have Google drive installed in your macos? If yes try to quit it "completely" and then try again. if not Google drive,check Dropbox or OneDrive. Some applications will lock on to the usb port once it detects a usb drive. See if any of the active app requires usb access, that might could be causing this issue.

 3) try disabling the hotplug detection from settings in OpenMTP.
 4) make sure that you have enabled file transfer and accepted "enable mtp data access" on your Samsung phone
 5) See if the alpha version of the app works for you. Here's a link to the latest test flight build of OpenMTP

 Apple silicone: https://github.com/ganeshrvel/openmtp-testflight/releases/download/v3.2.0-alpha.5/openmtp-3.2.0-alpha.5-mac-arm64.dmg

Intel: https://github.com/ganeshrvel/openmtp-testflight/releases/download/v3.2.0-alpha.5/openmtp-3.2.0-alpha.5-mac-x64.dmg

Installation instructions for openmtp Alpha version: https://github.com/ganeshrvel/openmtp/issues/220#issuecomment-1214261070

Thanks again,
Ganesh

**Comment by somsak** [comment]
Stopping Dropbox did the trick!. 

I wonder why this didn't happen before Ventura released?

**Comment by hungchieu** [comment]
> Stopping Dropbox did the trick!.
> 
> I wonder why this didn't happen before Ventura released?

I don't have dropbox. I have onedrive. Disabled onedrive still not working.

**Comment by hungchieu** [comment]
> Could try these?
> 
> 1. Reboot your Mac (IMPORTANT). Simply restarting the machine has helped many.
> 2. Do you have Google drive installed on your MacBook? If yes, try to quit it "completely" and then try launching openmtp again. See if Dropbox or OneDrive is installed and quit them completely. Some applications will lock on to the usb port once it detects a usb drive. See if any of the active apps require usb access which might be causing this issue.
> 3. Try disabling the hotplug detection from settings in OpenMTP.
> 4. Make sure that you have enabled file transfer and accepted "enable mtp data access" on your Samsung phone
> 5. See if the alpha version of the app works for you. Here's a link to the latest test flight build of OpenMTP
> 
> Apple silicone: https://github.com/ganeshrvel/openmtp-testflight/releases/download/v3.2.0-alpha.5/openmtp-3.2.0-alpha.5-mac-arm64.dmg
> 
> Intel: https://github.com/ganeshrvel/openmtp-testflight/releases/download/v3.2.0-alpha.5/openmtp-3.2.0-alpha.5-mac-x64.dmg
> 
> Installation instructions for openmtp Alpha version: [#220 (comment)](https://github.com/ganeshrvel/openmtp/issues/220#issuecomment-1214261070)

Hi,
I've follow your instruction. Disabled hotplug and then the app won't crash. However, it still did not detect the phone. I've already tried "Transferring files" and "Transferring images" modes, in both Kalam mode and legacy mode.
I know Ventura is the problem. There are already report that Android File Transfer is not working since Ventura beta.

**Comment by hungchieu** [comment]
> * After disabling the "USB hotplug" tap on the refresh icon to list the files
> * if you have Google drive/dropbox quit them completely
> * Reboot your machine
> * Uninstall any drivers that were installed by Samsungs SmartSwitch (if you had used it previously). More info: [Samsung Galaxy S20+ is not recognized #270](https://github.com/ganeshrvel/openmtp/issues/270)

Good news. It works.
I don't have Samsung SmartSwitch.
Currently is on Kalam mode, USB hotplug disabled, phone selected "Transferring files".
Each refresh will make the phone prompt for allow accessing files. I choose "Allow", after several attempts, it works.

**Comment by oldman20** [comment]
> > Could try these?
> > 
> > 1. Reboot your Mac (IMPORTANT). Simply restarting the machine has helped many.
> > 2. Do you have Google drive installed on your MacBook? If yes, try to quit it "completely" and then try launching openmtp again. See if Dropbox or OneDrive is installed and quit them completely. Some applications will lock on to the usb port once it detects a usb drive. See if any of the active apps require usb access which might be causing this issue.
> > 3. Try disabling the hotplug detection from settings in OpenMTP.
> > 4. Make sure that you have enabled file transfer and accepted "enable mtp data access" on your Samsung phone
> > 5. See if the alpha version of the app works for you. Here's a link to the latest test flight build of OpenMTP
> > 
> > Apple silicone: https://github.com/ganeshrvel/openmtp-testflight/releases/download/v3.2.0-alpha.5/openmtp-3.2.0-alpha.5-mac-arm64.dmg
> > Intel: https://github.com/ganeshrvel/openmtp-testflight/releases/download/v3.2.0-alpha.5/openmtp-3.2.0-alpha.5-mac-x64.dmg
> > Installation instructions for openmtp Alpha version: [#220 (comment)](https://github.com/ganeshrvel/openmtp/issues/220#issuecomment-1214261070)
> 
> Hi, I've follow your instruction. Disabled hotplug and then the app won't crash. However, it still did not detect the phone. I've already tried "Transferring files" and "Transferring images" modes, in both Kalam mode and legacy mode. Legacy mode not detecting at all. Kalam mode give attached error. The phone does ask for permission couple times, I did pressed "Allow" every single time it ask. Not helping. I know Ventura is the problem. There are already report that Android File Transfer is not working since Ventura beta. ![Screenshot 2022-11-02 at 9 48 31 pm](https://user-images.githubusercontent.com/72768451/199522135-ee0d9495-0508-4512-89ea-e33da122901a.jpg)

alpha version or 3.2.1 pre-release force close too on Ventura 13.0.1, i tried after fresh install, none google driver or anything

**Comment by ganeshrvel** [comment]
Do you have preview app open on Ventura? If  yes quit it. 
If you have any video editing app, try quiting it as well.

**Comment by oldman20** [comment]
> Do you have preview app open on Ventura? If yes quit it.
> If you have any video editing app, try quiting it as well.

no, haven't preview app or any video editing app, still fc when open

**Comment by raveslave** [comment]
has everyone filed bug reports to dropbox, google-drive etc? would be great

@ganeshrvel normally this shouldn't be an issue right as Multi-Session MTP should be supported by most phones, correct!?

I guess its gonna be more tricky if dropbox gets an exclusive lock on the whole usb-device, didn't even know that was allowed in IOKit as the phone / camera might run composite device mode for other features that then also would be blocked... 

Is there a way in IOKit to detect the lock and release it?
